### ISSUE TITLE
Adding tabs when starting from zero tabs causes LateInitializationError

### ISSUE THREAD

**Issue by YahavFes** [flutter/flutter#68962]
After post this question in [Stackoverflow](https://stackoverflow.com/questions/64516441/flutter-error-occurred-when-add-tab-to-the-page?noredirect=1#comment114080502_64516441), it's seems that the issue I am facing might be a bug.

I have the following code that should display page with button to add tab in the page:

Edit (@shihaohong): I updated the sample to contain a minimal reproducible app. You will also need to turn on null-safety in your Flutter app on the `master` branch to see this error. You can do so by using `flutter run --enable-experiment=non-nullable`.

```flutter
import 'package:flutter/material.dart';
void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> with TickerProviderStateMixin {
  List<String> tabTitles = [];
  List<Widget> tabContents = [];
  late TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: tabContents.length, vsync: this);
  }

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  void _onTabAdd(int index) {
    setState(() {
      tabTitles.add("Tab ${tabTitles.length}");
      tabContents.add(
        Container(
          color: Colors.red,
          height: 200,
          width: 200,
        ),
      );
      _tabController = TabController(length: tabContents.length, vsync: this);
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('DataTable of Gene Lists'),
          centerTitle: true,
          actions: [
            FlatButton(
              child: Text('+'),
              onPressed: () => _onTabAdd(_tabController.index),
            ),
          ],
          bottom: PreferredSize(
            preferredSize: Size.fromHeight(40.0),
            child: Expanded(
              child: TabBar(
                controller: _tabController,
                tabs: tabTitles
                  .map((title) => Tab(text: title))
                  .toList(),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

When I click this button the content seems to load successfully, but the problem is that the title is not shown, and the following error is displayed as you can see in the picture:
![Philadelphia's Magic Gardens. This place was so cool!](https://i.stack.imgur.com/O0dF0.png)

It's important to tell that it happened after I updated to the last version of Flutter (before it worked with no problem).
I don't know what was the previous version, but the current version is:
```
Flutter 1.24.0-4.0.pre.51 • channel master • https://github.com/flutter/flutter.git
Framework • revision 720dff6a94 (9 hours ago) • 2020-10-24 03:57:03 -0400
Engine • revision 432974a52b
Tools • Dart 2.11.0 (build 2.11.0-251.0.dev)
```

**Comment by shihaohong** [comment]
It turns out that this issue happens when you initialize your application to have zero tabs. This is happening because the `_IndicatorPainter` code assumes that the old `IndicatorPainter` will have a list of tab offset values and text direction values, which is set upon performing layout. However, layout and painting for the tab bar never occurs when there are zero tabs. 

This is actually related to https://github.com/flutter/flutter/issues/40014 as well. I'll try to come up with a fix.

**Comment by goderbauer** [comment]
@shihaohong Is https://github.com/flutter/flutter/issues/68586 caused by the same bug and fixed by your PR?

**Comment by shihaohong** [comment]
Yes, I just verified it. Once that PR is merged, #68586 should be fixed as well.
